core/src/test/java/org/apache/calcite/util/Smalls.java

```diff
@@ -843,6 +843,30 @@ private void abc(StringBuilder sb, Object s) {
     }
   }
 
+  /** User-defined table-macro function with named and optional parameters. */
+  public static class AnotherTableMacroFunctionWithNamedParameters {
+    public TranslatableTable eval(
+        @Parameter(name = "R", optional = true) String r,
+        @Parameter(name = "S") String s,
+        @Parameter(name = "T", optional = true) Integer t,
+        @Parameter(name = "S2", optional = true) String s2) {
+      final StringBuilder sb = new StringBuilder();
+      abc(sb, r);
+      abc(sb, s);
+      abc(sb, t);
```

Comment:
How can we distinguish "View\"(r=>'6', s=>'5')) since both `TableMacroFunctionWithNamedParameters`
 and `AnotherTableMacroFunctionWithNamedParameters` match?

Reply:
In this case, will be used the first registered function. Here is the code which returns it: https://github.com/apache/calcite/blob/1baee8524a8daa9c67f08f83080e269fc5938bc5/core/src/main/java/org/apache/calcite/sql/SqlUtil.java#L451

Reply:
Without your patch, will it happen too?

Reply:
Yes, the behavior would be the same without these changes.